File: parent_app/lib/models/screen_time_rule.dart
/// Screen-time rule for a child (UI model). Times are "minutes from midnight"
/// (0..1439) so they're timezone-agnostic and easy to compare on-device later.
class ScreenTimeRule {
  int dailyLimitMinutes;
  bool bedtimeEnabled;
  int bedtimeStart; // minutes from midnight
  int bedtimeEnd;
  bool paused;

  ScreenTimeRule({
    this.dailyLimitMinutes = 120,
    this.bedtimeEnabled = false,
    this.bedtimeStart = 21 * 60, // 21:00
    this.bedtimeEnd = 7 * 60, // 07:00
    this.paused = false,
  });

  ScreenTimeRule copy() => ScreenTimeRule(
        dailyLimitMinutes: dailyLimitMinutes,
        bedtimeEnabled: bedtimeEnabled,
        bedtimeStart: bedtimeStart,
        bedtimeEnd: bedtimeEnd,
        paused: paused,
      );

  factory ScreenTimeRule.fromMap(Map<String, dynamic> map) => ScreenTimeRule(
        dailyLimitMinutes:
            (map['dailyLimitMinutes'] is int) ? map['dailyLimitMinutes'] : 120,
        bedtimeEnabled: map['bedtimeEnabled'] == true,
        bedtimeStart: (map['bedtimeStart'] is int) ? map['bedtimeStart'] : 21 * 60,
        bedtimeEnd: (map['bedtimeEnd'] is int) ? map['bedtimeEnd'] : 7 * 60,
        paused: map['paused'] == true,
      );

  Map<String, dynamic> toMap() => {
        'dailyLimitMinutes': dailyLimitMinutes,
        'bedtimeEnabled': bedtimeEnabled,
        'bedtimeStart': bedtimeStart,
        'bedtimeEnd': bedtimeEnd,
        'paused': paused,
      };
}

File: child_windows/src/GuardNest.Core/Rules/ScreenTimeRule.cs
namespace GuardNest.Core.Rules;

/// <summary>
/// The family's screen-time rule. Minutes-from-midnight matches the parent app's
/// model, so both sides speak the same language.
/// </summary>
public sealed record ScreenTimeRule(
    int DailyLimitMinutes = 120,
    bool BedtimeEnabled = false,
    int BedtimeStart = 21 * 60,
    int BedtimeEnd = 7 * 60,
    bool Paused = false);

/// <summary>Blocked executables, merged from the family-wide and per-child rules.</summary>
public sealed record AppRules(IReadOnlySet<string> Blocked)
{
    public static readonly AppRules Empty =
        new(new HashSet<string>(StringComparer.OrdinalIgnoreCase));

    public bool IsBlocked(string executable) => Blocked.Contains(executable);
}
